Automattic/jetpack
# Forms: Remove Salesforce integration from third party files

Fixes https://github.com/Automattic/jpop-issues/issues/9331

## Proposed changes:

* This PR removes the third party Salesforce integration, to prevent duplicate Salesforce form submissions.
* It looks like the Salesforce integration was made within the forms package here - https://github.com/Automattic/jetpack/pull/30191- in order to replace the third party integration added here - https://github.com/Automattic/jetpack/pull/26903.


### Other information:


## Jetpack product discussion

https://github.com/Automattic/jpop-issues/issues/9331

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Replicating the issue:

* Add a form block to a test site, and choose the Salesforce Lead Form variation.
* In the block settings sidebar, add an organization ID. You don't need a Salesforce account for this, but you can get a free 30 day trial at salesforce.com if preferred. Otherwise, adding a dummy organization ID will work, such as `TESTINGONETWOTHREE-1234`.
* Add breakpoints, or just `error_log` calls, at the points mentioned in the issue linked above. Alternatively, add a snippet (eg. using a Code Snippets plugin) to hook into `pre_http_request`, to listen to requests made with `wp_remote_post`:
```
add_filter('pre_http_request', function ($pre, $parsed_args, $url) {
    if ($parsed_args['method'] === 'POST') {
        error_log('wp_remote_post called to URL: ' . $url);
        error_log('Request Args: ' . print_r($parsed_args, true));
    }
    return $pre;
}, 10, 3);
```
* You may need to comment out the `if ( $is_spam ) {` checks in both methods if Akismet is enabled.
* On the front-end, add a submission
* Notice in your error logs that there will be two submissions to the Salesforce URL - `webto.salesforce.com....` (or two entries in the corresponding files if adding debugging).

To test the fix:

* Apply this PR, and add the same debugging as in the earlier step.
* Add a submission (making sure the details are different), and notice that there should only be one submission to the 
`webto.salesforce.com....` URL.


WoA/Simple:
* I had difficulty testing on Simple / WoA, but had the same results before and after (no duplications before on WoA)
